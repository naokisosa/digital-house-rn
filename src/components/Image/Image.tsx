import { StyleSheet, Image as NativeImage, ImageSourcePropType, ViewStyle } from 'react-native';
import React from 'react';

const IMAGE_PLACEHOLDER = require('@assets/placeholder.png');

type ImageSizes = 'big' | 'small';

interface Props {
  size?: ImageSizes;
  source?: ImageSourcePropType;
  style?: ViewStyle;
}

const IMAGE_SIZES = new Map<ImageSizes, number>([
  ['big', 200],
  ['small', 55],
]);

const Image = ({ size = 'big', source = IMAGE_PLACEHOLDER, style }: Props) => {
  return <NativeImage source={source} style={[styles.image(size), style]} />;
};

export default Image;

const styles = StyleSheet.create<any>({
  image: (size: ImageSizes) => ({
    width: IMAGE_SIZES.get(size),
    height: IMAGE_SIZES.get(size),
  }),
});
