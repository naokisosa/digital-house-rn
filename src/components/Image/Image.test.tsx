import 'react-native';
import React from 'react';

import Image from './Image';
import renderer from 'react-test-renderer';
import { SINGLE_PRODUCT } from '@mocks';

describe('Test Image component', () => {
  it('renders correctly with default values', () => {
    const { toJSON } = renderer.create(<Image />);
    expect(toJSON()).toMatchSnapshot();
  });
  it('renders correctly with size="big"', () => {
    const { toJSON } = renderer.create(<Image size="big" />);
    expect(toJSON()).toMatchSnapshot();
  });
  it('renders correctly with size="small"', () => {
    const { toJSON } = renderer.create(<Image size="small" />);
    expect(toJSON()).toMatchSnapshot();
  });
  it('renders correctly when providing external uri', () => {
    const { toJSON } = renderer.create(<Image source={{ uri: SINGLE_PRODUCT.image }} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
