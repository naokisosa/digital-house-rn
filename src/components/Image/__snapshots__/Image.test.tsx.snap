// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Image component renders correctly when providing external uri 1`] = `
<View
  style={
    [
      {
        "overflow": "hidden",
      },
      [
        {
          "height": 200,
          "width": 200,
        },
        undefined,
      ],
    ]
  }
>
  <FastImageView
    defaultSource={
      {
        "testUri": "../../../src/assets/placeholder.png",
      }
    }
    resizeMode="cover"
    source={
      {
        "uri": "https://loremflickr.com/640/480/city",
      }
    }
    style={
      {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
</View>
`;

exports[`Test Image component renders correctly with default values 1`] = `
<View
  style={
    [
      {
        "overflow": "hidden",
      },
      [
        {
          "height": 200,
          "width": 200,
        },
        undefined,
      ],
    ]
  }
>
  <FastImageView
    defaultSource={
      {
        "testUri": "../../../src/assets/placeholder.png",
      }
    }
    resizeMode="cover"
    source={
      {
        "testUri": "../../../src/assets/placeholder.png",
      }
    }
    style={
      {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
</View>
`;

exports[`Test Image component renders correctly with size="big" 1`] = `
<View
  style={
    [
      {
        "overflow": "hidden",
      },
      [
        {
          "height": 200,
          "width": 200,
        },
        undefined,
      ],
    ]
  }
>
  <FastImageView
    defaultSource={
      {
        "testUri": "../../../src/assets/placeholder.png",
      }
    }
    resizeMode="cover"
    source={
      {
        "testUri": "../../../src/assets/placeholder.png",
      }
    }
    style={
      {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
</View>
`;

exports[`Test Image component renders correctly with size="small" 1`] = `
<View
  style={
    [
      {
        "overflow": "hidden",
      },
      [
        {
          "height": 55,
          "width": 55,
        },
        undefined,
      ],
    ]
  }
>
  <FastImageView
    defaultSource={
      {
        "testUri": "../../../src/assets/placeholder.png",
      }
    }
    resizeMode="cover"
    source={
      {
        "testUri": "../../../src/assets/placeholder.png",
      }
    }
    style={
      {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
</View>
`;
